File: 백준/Silver/10866. 덱/덱.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		Deque<Integer> dq = new LinkedList<>();

		int n = Integer.parseInt(br.readLine());

		while (n-- > 0) {
			String s = br.readLine();
			StringTokenizer st = new StringTokenizer(s);

			String input = st.nextToken();

			if (input.equals("push_back")) {
				int num = Integer.parseInt(st.nextToken());

				dq.addLast(num);
			} else if (input.equals("push_front")) {
				int num = Integer.parseInt(st.nextToken());

				dq.addFirst(num);
			} else if (input.equals("pop_front")) {
				if (dq.isEmpty())
					sb.append("-1");
				else {
					sb.append(dq.peekFirst());
					dq.removeFirst();
				}
				sb.append("\n");

			} else if (input.equals("pop_back")) {
				if (dq.isEmpty())
					sb.append("-1");
				else {
					sb.append(dq.peekLast());
					dq.removeLast();
				}
				sb.append("\n");
			} else if (input.equals("size")) {
				sb.append(dq.size()).append("\n");
			} else if (input.equals("empty")) {
				if (dq.isEmpty())
					sb.append("1");
				else
					sb.append("0");
				sb.append("\n");
			} else if (input.equals("front")) {
				if (dq.isEmpty())
					sb.append("-1");
				else
					sb.append(dq.peekFirst());
				sb.append("\n");
			} else {
				if (dq.isEmpty())
					sb.append("-1");
				else
					sb.append(dq.peekLast());
				sb.append("\n");
			}
		}

		System.out.println(sb);
	}
}

File: 백준/Silver/10866. 덱/덱.cc
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int n;
deque<int> dq;
string cmd;
int num;

int main() {
    ios_base::sync_with_stdio(0); cin.tie(0);

    cin >> n;

    while (n--) {
        cin >> cmd;
        if (cmd == "push_back") {
            cin >> num;
            dq.push_back(num);
            continue;
        } else if (cmd == "push_front") {
            cin >> num;
            dq.push_front(num);
            continue;
        } else if (cmd == "pop_back") {
            if (!dq.empty()) {
                cout << dq.back();
                dq.pop_back();
            }
            else cout << -1;
        } else if (cmd == "pop_front") {
            if (!dq.empty()) {
                cout << dq.front();
                dq.pop_front();
            }
            else cout << -1;
        } else if (cmd == "size") {
            cout << dq.size();
        } else if (cmd == "empty") {
            if (dq.empty()) cout << 1;
            else cout << 0;
        } else if (cmd == "front") {
            if (!dq.empty()) cout << dq.front();
            else cout << -1;
        } else if (cmd == "back") {
            if (!dq.empty()) cout << dq.back();
            else cout << -1;
        }
        cout << "\n";
    }

    return 0;
}